src/pointer-analysis/dereference.cpp

```diff
@@ -612,6 +612,64 @@ bool dereferencet::dereference_type_compare(
   return false;
 }
 
+void dereferencet::check_pointer_alignment(
+  modet mode,
+  const type2tc &type,
+  const expr2tc &deref_expr,
+  const guardt &guard)
+{
+  if (options.get_bool_option("no-align-check"))
+    return;
```

Comment:
Technically that check is more or less redundant, now that check_alignment also has that function call.

Reply:
Thanks, @intrigus-lgtm. I agree with you. I have just removed it here https://github.com/esbmc/esbmc/commit/d48ca77b2a53f80909a767c165d19da596494f13.

I'll address the other comments via small PRs.